src/Formik.tsx

```diff
@@ -12,6 +12,13 @@ import {
   setNestedObjectValues,
 } from './utils';
 
+export type Extends<A, B> = A extends B ? true : false;
+export type Tree<Children, Leaf> = {
+  [K in keyof Children]?: Children[K] extends object
+    ? false extends Extends<Children[K], object> ? any : Tree<Children[K], Leaf>
```

Comment:
The extra check is to support when `Children` is `any`, because `any extends object ? A : B` is `A | B`.